### PR TITLE
⬆️ Upgrade Flow, Babel, Eslint and related deps

### PR DESCRIPTION
**Pull request by josemigallas**

### CODE REVIEW

File: spec/javascripts/PaymentGateways/braintree/BraintreeForm.spec.jsx [hunk 1]
```diff
@@ -1,9 +1,24 @@
+// @flow
```
Comment (damianpm) [inline spec/javascripts/PaymentGateways/braintree/BraintreeForm.spec.jsx:1]: Are this changes related to updating packages? Maybe they belong to another PR (refactor)

Comment (josemigallas) [reply spec/javascripts/PaymentGateways/braintree/BraintreeForm.spec.jsx:1]: The upgrade make these tests fail, not sure why, but I think we should merge it together.

**Review by hallelujah — APPROVED** [review]

File: spec/javascripts/PaymentGateways/braintree/BraintreeForm.spec.jsx (hunk 1 above)
Comment (damianpm) [reply spec/javascripts/PaymentGateways/braintree/BraintreeForm.spec.jsx:1]: OK, let's merge

**Review by damianpm — APPROVED** [review]
